src/Error/PhpError.php

```diff
@@ -0,0 +1,191 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.4.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Error;
+
+/**
+ * Object wrapper around PHP errors that are emitted by `trigger_error()`
+ */
+class PhpError
+{
+    /**
+     * @var int
+     */
+    private $code;
+
+    /**
+     * @var string
+     */
+    private $message;
+
+    /**
+     * @var string|null
+     */
+    private $file;
+
+    /**
+     * @var int|null
+     */
+    private $line;
+
+    /**
+     * Stack trace data. Each item should have a `reference`, `file` and `line` keys.
+     *
+     * @var array<array<string, int>>
+     */
+    private $trace;
+
+    /**
+     * @var array<int, string>
+     */
+    private $levelMap = [
+        E_PARSE => 'error',
+        E_ERROR => 'error',
+        E_CORE_ERROR => 'error',
+        E_COMPILE_ERROR => 'error',
+        E_USER_ERROR => 'error',
+        E_WARNING => 'warning',
+        E_USER_WARNING => 'warning',
+        E_COMPILE_WARNING => 'warning',
+        E_RECOVERABLE_ERROR => 'warning',
+        E_NOTICE => 'notice',
+        E_USER_NOTICE => 'notice',
+        E_STRICT => 'strict',
+        E_DEPRECATED => 'deprecated',
+        E_USER_DEPRECATED => 'deprecated',
+    ];
+
+    /**
+     * @var array<string, int>
+     */
+    private $logMap = [
+        'error' => LOG_ERR,
+        'warning' => LOG_WARNING,
+        'notice' => LOG_NOTICE,
+        'strict' => LOG_NOTICE,
+        'deprecated' => LOG_NOTICE,
+    ];
+
+    /**
+     * Constructor
+     *
+     * @param int $code The PHP error code constant
+     * @param string $message The error message.
+     * @param string|null $file The filename of the error.
+     * @param int|null $line The line number for the error.
+     * @param array $trace The backtrace for the error.
+     */
+    public function __construct(
+        int $code,
+        string $message,
+        ?string $file = null,
+        ?int $line = null,
+        array $trace = [],
+    ) {
+        $this->code = $code;
+        $this->message = $message;
+        $this->file = $file;
+        $this->line = $line;
+        $this->trace = $trace;
+    }
+
+    /**
+     * Get the PHP error constant.
+     *
+     * @return string
+     */
+    public function getCode(): int
```

Comment:
Should this be `code()` instead of `getCode()`?

Reply:
It could be. I was thinking it could be more ergonomic to emulate the existing methods of `Exception` so that users have to remember fewer methods names if they ever need to interact with error and exception handlers.

Reply:
I assumed you were following Exception. I'd like to start promoting more of these style guides in Cake in the future although modifying anything existing would be really annoying.

Arbitrary using get or or not starts to get messy. Something to discuss.

Reply:
That's fair I can put together an update to the internal development docs on method naming conventions and when to diverge from them.